File: Shop.API/Core/Models/Order.cs
using System;
using System.ComponentModel.DataAnnotations.Schema;

namespace Shop.API.Core.Models
{
	public class Order
	{
		public int Id { get; set; }
		public string Items { get; set; }

		[Column(TypeName = "decimal(18, 2)")]
		public decimal Price { get; set; }
		public string Reference { get; set; }
		public string Address { get; set; }
		public string PhoneNumber { get; set; }
		public string PhoneNumber2 { get; set; }
		public string City { get; set; }
		public string State { get; set; }
		public User User { get; set; }
		public int UserId { get; set; }
		public DateTime OrderDate { get; set; }
		public bool IsShipped { get; set; }
		public DateTime ShippingDate { get; set; }
	}
}

File: Shop.SPA/src/app/_models/Order.ts
import { User } from './User';

export interface Order {
  items?: any;
  totalPrice?: number;
  address?: string;
  city?: string;
  state?: string;
  reference?: any;
  phoneNumber?: string;
  phoneNumber2?: string;
}
export interface OrderForListAdmin {
  totalPrice?: number;
  orderDate?: Date;
  isShipped?: boolean;
  reference?: string;
}
export interface OrderViewAdmin {
  id: number;
  user: User;
  orderDate?: Date;
  shippingDate?: Date;
  isShipped?: boolean;
  items?: any;
  totalPrice?: number;
  address?: string;
  city?: string;
  state?: string;
  reference?: string;
  phoneNumber?: string;
  phoneNumber2?: string;
}
